lib/active_snapshot/models/snapshot.rb

```diff
@@ -145,5 +145,80 @@ def fetch_reified_items(readonly: true)
       return [reified_parent, reified_children_hash]
     end
 
+    def diff(new_snapshot)
```

Comment:
I really like this method however I think that it increases the perceived complexity of the gem quite a bit. Maybe we should move it into a separate file/concern?

Reply:
agreed, it was great how easy it was to understand the source code